processor/probabilisticsamplerprocessor/tracesprocessor.go

```diff
@@ -174,6 +174,7 @@ func (tp *traceProcessor) processTraces(ctx context.Context, td ptrace.Traces) (
 	td.ResourceSpans().RemoveIf(func(rs ptrace.ResourceSpans) bool {
 		rs.ScopeSpans().RemoveIf(func(ils ptrace.ScopeSpans) bool {
 			ils.Spans().RemoveIf(func(s ptrace.Span) bool {
+				tp.telemetryBuilder.ProcessorProbabilisticSamplerCountSpansSampled.Add(ctx, 1)
```

Comment:
I am not sure I get this - did you mean to record a count of all spans going through the processor?

Reply:
Yes. As I understand the issue linked this is what was requested.

Reply:
Hi @atoulme , let me know if I am missing something here?

Reply:
This won't help, unfortunately. You should either store the result of `commonShouldSampleLogic` and add 1 in case it is being sampled, or add the counter directly to the `commonShouldSampleLogic` function itself. As it is, it's just counting the number of spans passing through the processor, as @atoulme pointed out, which doesn't help the original use case: they need a way to find how many spans were sampled.

Reply:
I'd also ask you to add a test for this behavior.